CI failure in test_reactor.py more frequent since dependency updates.

## Describe the bug

Since updating our dependencies after the latest release, I've noticed a failure in the CI relatively frequently, certainly more than before the update.

The failure  is on a comparison to 0.0, which fails as occasionally it is larger than 1e-12 in absolute terms. This is not quite machine precision, but does involve summing several integrations which have a tolerance probably slightly higher than machine precision, hence the occasional failure.

I cannot explain why it is more frequent than before, but could relate to some integration changes in low-level dependencies (e.g. scipy, numpy).

## Steps to reproduce

Steps to reproduce the behaviour:
Run the CI on a branch enough times and you will eventually encounter:
```python

self = <EUDEMO_tests.test_reactor.TestEUDEMO object at 0x7fb97619aa30>

    def test_tf_build(self):
        """
        Test the results of the TF build.
        """
        tf_builder: TFCoilsBuilder = self.reactor.get_builder(
            EUDEMOReactorDesign.TF_COILS
        )
        assert tf_builder is not None
        assert tf_builder.design_problem is not None
    
        tf_component: TFCoilsComponent = self.component.get_component(
            EUDEMOReactorDesign.TF_COILS
        )
        assert tf_component is not None
    
        # Check field at origin
        field = tf_component.field(
            self.reactor.params.R_0.value, 0.0, self.reactor.params.z_0.value
        )
        assert field is not None
>       assert field == pytest.approx([0, -5.0031, 0])
E       assert array([-1.131...52450577e-14]) == approx([0 ± 1... 0 ± 1.0e-12])
E         comparison failed. Mismatched elements: 1 / 3:
E         Max absolute difference: 1.1312617509418033e-12
E         Max relative difference: 1.0
E         Index | Obtained                | Expected   
E         0     | -1.1312617509418033e-12 | 0 ± 1.0e-12

examples/design/EU-DEMO/EUDEMO_tests/test_reactor.py:128: AssertionError
```

## Expected behaviour
This test should never fail.

## Screenshots / Tracebacks

If applicable, add screenshots / tracebacks to help explain your problem.

## Environment (please complete the following information)

- OS: CI on develop

## Additional context

Add any other context about the problem here.

--- a/examples/design/EU-DEMO/EUDEMO_tests/test_reactor.py
+++ b/examples/design/EU-DEMO/EUDEMO_tests/test_reactor.py
@@ -125,7 +125,7 @@ class TestEUDEMO:
             self.reactor.params.R_0.value, 0.0, self.reactor.params.z_0.value
         )
         assert field is not None
-        assert field == pytest.approx([0, -5.0031, 0])
+        assert field == pytest.approx([0, -5.0031, 0], rel=1e-6, abs=2e-12)
 
     def test_tf_save(self):
         """

